### PR TITLE
Issue6 - User-creation endpoint - tenzin

### LINKED ISSUE #6: Registration Process - Backend
## Story

Design the front facing Sign-in/Sign Up page 

**As a Trainee/ Volunteer I Want to land on the Sign-in/Sign Up page and make the selection so that I can access my account or create a new account, depending on my selection a registration form will pop up. 

## Acceptance Criteria

1. Sign in option provided, and able to be selected
2. Sign up option provided, and able to be selected
3. Appropriate form generated depending on selection made

## Implementation Notes

1. Implement React component

### PR DESCRIPTION
**Pull request by tenzyns**
## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

user-creation endpoint is created using POST request.

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #6 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] I have commented my code
- [x] I have updated any documentation

### CODE REVIEW

File: server/api.js [hunk 1]
```diff
@@ -1,9 +1,59 @@
...
+		const newUser = {
+			firstName: req.body.firstName,
+			lastName: req.body.lastName,
+			username: req.body.username,
+			cohortId: req.body.cohortId,
+			password: hashedPassword,
+		};
+		//checking if username already exist
+		pool
+		.query("SELECT user_name FROM users WHERE user_name=$1", [newUser.username])
```
Comment (Professional-Kate) [inline server/api.js:24]: This line is perfectly fine, but there is this: ``SELECT EXISTS(SELECT user_name FROM users WHERE user_name = $1);``

The syntax could be wrong, so you may want to research it. But this returns a boolean value based on the expression in the brackets. If it exists you'll get a TRUE returned else FALSE. But this line is perfectly fine it does the same thing haha

File: server/api.js [hunk 2]
```diff
@@ -1,9 +1,59 @@
...
+		.then((result) => {
+			if (result.rows.length > 0) {
+				res.status(400).send("Username already exist!");
+			} else {
+				//Checking if cohort ID exist
+				pool.query("SELECT id FROM cohorts WHERE id=$1", [newUser.cohortId])
```
Comment (Professional-Kate) [inline server/api.js:30]: so this here wouldn't be needed. In the frontend signup form the cohorts dropdown will be populated from a GET request from the /cohorts endpoint. Every cohort will be a valid one - for sure, that's what I tried to explain last night very badly lmao

File: server/api.js [hunk 3]
```diff
@@ -1,9 +1,59 @@
...
+				.then((result1) => {
+					if (result1.rows.length < 1) {
+						res.status(400).send("Cohort ID doesn't exist");
+					//ensuring password and username length is 6 or more characters
+					} else if (req.body.password.length < 6 || newUser.username.length < 6) {
+						res.status(400).send("Password or username must be of 6 or more characters!");
+						//ensuring all fields are completed
+					} else if (newUser.firstName && newUser.lastName && newUser.cohortId) {
+						const query = "INSERT INTO users (first_name, last_name, pass_hash, user_name, cohort_id) VALUES ($1, $2, $3, $4, $6)";
```
Comment (Professional-Kate) [inline server/api.js:39]: might want to change the $ signs at the end. Not sure if it'll break but you're missing ``$5``, it goes from 4 to 6

**Review by Professional-Kate — CHANGES_REQUESTED** [review]
Just a couple things that need changing. Otherwise, really nice work - you've done a wonderful job

File: server/api.js (hunk 2 above)
Comment (tenzyns) [reply server/api.js:30]: This is done.

**Review by Professional-Kate — APPROVED** [review]
LGTM!

**Pull request merged**